autogalaxy/analysis/model_util.py

```diff
@@ -0,0 +1,362 @@
+    data which are fit poorly.
+
+    The hyper model is typically used in pipelines to refine and improve an `Inversion` after model-fits that fit the
+    `Galaxy` light and mass components.
+
+    Parameters
+    ----------
+    setup_hyper : SetupHyper
+        The setup of the hyper analysis if used (e.g. hyper-galaxy noise scaling).
+    result : af.Result
+        The result of a previous `Analysis` search whose maximum log likelihood model forms the basis of the hyper model.
+    include_hyper_image_sky : hd.HyperImageSky
+        This must be true to include the hyper-image sky in the model, even if it is turned on in `setup_hyper`.
+
+    Returns
+    -------
+    af.Collection
+        The hyper model, which has an instance of the input results maximum log likelihood model with certain hyper
+        model components now free parameters.
+    """
+
+    model = result.instance.as_model((pix.Pixelization, reg.Regularization))
+
+    if setup_hyper is None:
+        return None
+
+    if setup_hyper.hyper_galaxy_names is None:
+        if not has_pixelization_from_model(model=model):
+            if setup_hyper.hypers_all_off:
+                return None
+            if setup_hyper.hypers_all_except_image_sky_off:
+                if not include_hyper_image_sky:
+                    return None
+
+    model.hyper_image_sky = setup_hyper.hyper_image_sky
+    model.hyper_background_noise = setup_hyper.hyper_background_noise
+
+    if setup_hyper.hyper_galaxy_names is not None:
+
+        for path_galaxy, galaxy in result.path_galaxy_tuples:
+            if path_galaxy[-1] in setup_hyper.hyper_galaxy_names:
+                if not np.all(result.hyper_galaxy_image_path_dict[path_galaxy] == 0):
+
+                    galaxy = getattr(model.galaxies, path_galaxy[-1])
+
+                    setattr(galaxy, "hyper_galaxy", af.Model(g.HyperGalaxy))
+
+    return model
+
+
+def hyper_fit(hyper_model: af.Collection, setup_hyper, result: af.Result, analysis):
+    """
+    Perform a hyper-fit, which extends a model-fit with an additional fit which fixes the non-hyper components of the
+    model (e.g., `LightProfile`'s, `MassProfile`) to the `Result`'s maximum likelihood fit. The hyper-fit then treats
+    only the hyper-model components as free parameters, which are any of the following model components:
+
+    1) The `Pixelization` of any `Galaxy` in the model.
+    2) The `Regularization` of any `Galaxy` in the model.
+    3) Hyper data components like a `HyperImageSky` or `HyperBackgroundNoise` if input into the function.
+    4) `HyperGalaxy` components of the `Galaxy`'s in the model, which are used to scale the noise in regions of the
+    data which are fit poorly.
+
+    The hyper model is typically used in pipelines to refine and improve an `Inversion` after model-fits that fit the
+    `Galaxy` light and mass components.
+
+    Parameters
+    ----------
+    hyper_model : Collection

```

Comment:
There's not need to put type annotations in the doc string when they're already defined in the signature

Reply:
I think that depends on how Sphinx works.